src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -1224,83 +1224,144 @@ CATCH_RETURN();
+}
+
+// Method Description:
+// - Converts a float fontSize into the corresponding float value to create a DWRITE_FONT_AXIS_VALUE with
+// Arguments:
+// - fontSize: the old float value to be converted into an axis value
+// Return value:
+// - The float value corresponding to the passed in fontSize
+float CustomTextLayout::_DIPsToPoints(const float fontSize)
+{
+    return fontSize * (72.0f / 96.0f);
+}
+
 // Method Description:
 // - Fill any missing axis values that might be known but were unspecified, such as omitting
 //   the 'wght' axis tag but specifying the old DWRITE_FONT_WEIGHT enum
 // Arguments:
-// - todo: fill this out
-std::vector<DWRITE_FONT_AXIS_VALUE> CustomTextLayout::GetAxisVector(DWRITE_FONT_WEIGHT /*fontWeight*/,
-                                                                    DWRITE_FONT_STRETCH /*fontStretch*/,
-                                                                    DWRITE_FONT_STYLE /*fontStyle*/,
-                                                                    float /*fontSize*/,
-                                                                    ::Microsoft::WRL::ComPtr<IDWriteTextFormat3>& format)
+// - fontWeight: the old DWRITE_FONT_WEIGHT enum to be converted into an axis value
+// - fontStretch: the old DWRITE_FONT_STRETCH enum to be converted into an axis value
+// - fontStyle: the old DWRITE_FONT_STYLE enum to be converted into an axis value
+// - fontSize: the number to convert into an axis value
+// - format: the IDWriteTextFormat3 to get the defined axes from
+// Return value:
+// - The fully formed axes vector
+std::vector<DWRITE_FONT_AXIS_VALUE> CustomTextLayout::_GetAxisVector(const DWRITE_FONT_WEIGHT fontWeight,
+                                                                     const DWRITE_FONT_STRETCH fontStretch,
+                                                                     const DWRITE_FONT_STYLE fontStyle,
+                                                                     const float fontSize,
+                                                                     const ::Microsoft::WRL::ComPtr<IDWriteTextFormat3>& format)
 {
-    // todo: complete this function
-    //enum AxisTagPresence
-    //{
-    //    AxisTagPresenceNone = 0,
-    //    AxisTagPresenceWeight = 1,
-    //    AxisTagPresenceWidth = 2,
-    //    AxisTagPresenceItalic = 4,
-    //    AxisTagPresenceSlant = 8,
-    //    AxisTagPresenceOpticalSize = 16,
-    //};
+    enum AxisTagPresence
+    {
+        AxisTagPresenceNone = 0,
+        AxisTagPresenceWeight = 1,
+        AxisTagPresenceWidth = 2,
+        AxisTagPresenceItalic = 4,
+        AxisTagPresenceSlant = 8,
+        AxisTagPresenceOpticalSize = 16,
+    };
     const auto axesCount = format->GetFontAxisValueCount();
     std::vector<DWRITE_FONT_AXIS_VALUE> axesVector;
     axesVector.resize(axesCount);

```

Comment:
Doesn't resize allocate N items with the default constructor? Did you mean reserve?

Reply:
`reserve` doesn't work, I suspect its because we pass in `axesVector.data()` into the `GetFontAxisValues` call, and the array returned from `axesVector.data()` will have a size of 0 if there are no items in the vector (no matter how much space the vector has been allocated). 

So we could work around this by not having a vector at all and instead using an array, but we won't be able to use 
`axesCount` in that case since it is not a compile-time constant, we would need to do `DWRITE_FONT_AXIS_VALUE axesArray[5]`. Is that preferred?

Reply:
Actually wait no we can't even do `DWRITE_FONT_AXIS_VALUE axesArray[5]` because there might actually be more than 5 `DWRITE_FONT_AXIS_VALUE`s in there (the number of possible axes is limited by the _font_ being used)

Reply:
OK. Two things.

1. Sorry. Resize is correct if you're going to pass the data member in because it needs already constructed items to fill with data. I was confused.
2. If there are more than 5 items that `GetFontAxisValues` wants to return.... aren't we going to have a problem still because it was only resized to 5?

Reply:
We resize with the value returned from `GetFontAxisValueCount`, so we're safe there!